src/sources/soundsourcemp3.cpp

```diff
@@ -600,34 +638,38 @@ SINT SoundSourceMp3::readSampleFrames(
             const SINT madSynthOffset =
                     m_madSynth.pcm.length - m_madSynthCount;
             DEBUG_ASSERT(madSynthOffset < m_madSynth.pcm.length);
-            if (isChannelCountMono()) {
+            const SINT madSynthChannelCount = m_madSynth.pcm.channels;
+            DEBUG_ASSERT(0 < madSynthChannelCount);
+            DEBUG_ASSERT(madSynthChannelCount <= getChannelCount());
+#ifndef QT_NO_DEBUG_OUTPUT
+            if (madSynthChannelCount != getChannelCount()) {
+                qDebug() << "Reading MP3 data with different number of channels"
+                        << madSynthChannelCount << "<>" << getChannelCount();
+            }
+#endif
+            if (1 == math_min(madSynthChannelCount, getChannelCount())) {
                 for (SINT i = 0; i < synthReadCount; ++i) {
-                    const CSAMPLE sampleValue = madScale(
+                    const CSAMPLE sampleValue = madScaleSampleValue(
                             m_madSynth.pcm.samples[0][madSynthOffset + i]);
                     *pSampleBuffer = sampleValue;
                     ++pSampleBuffer;
-                    if (readStereoSamples) {
+                    if (readStereoSamples || isChannelCountStereo()) {
+                        // Duplicate the 1st channel
                         *pSampleBuffer = sampleValue;
                         ++pSampleBuffer;
                     }
                 }
-            } else if (isChannelCountStereo() || readStereoSamples) {
+            } else {
+                DEBUG_ASSERT(readStereoSamples || isChannelCountStereo());
```

Comment:
Not sure why readStereoSamples is allways 1 here? When is this function called with readStereoSamples = false? 
Only if  isChannelCountStereo() was false?